TACC-Cloud/hazmapper
# Task/wg 193 add tup core components lib

## Overview: ##

## PR Status: ##


## Related Jira tickets: ##

* [WG-193](https://tacc-main.atlassian.net/browse/WG-193)

## Summary of Changes: ##
- Added core-component dir from TUP-UI 
-  Updated configs and packages needed for core-Components
- Ignoring core-components tests in test script
- Allow for implict any type as per the tsconfigs of TUP
- Allow for js to be compiled
- Added styles README from TUP for our reference

## Testing Steps: ##
1.  Make sure you're using latest iron node by running nvm use lts/iron
2. Then npm ci
3. Then build and run dev
4. Go to http://localhost:4200/ and make sure it displays 
"Main Menu
Projects
You have {x} projects."
6. In the code base, go to src/pages/MainMenu.tsx change line 12 to (!isLoading) and make sure loading spinner appears. 
7.  In the code base, go to src/pages/MainMenu.tsx change line 19 to (!error) to make sure message displays in red. 

## Notes: ##
